packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -142,21 +181,28 @@ function ProviderOptionsScreen({ route, navigation }: Props) {
     provider.onSelected()
   }
 
-  return (
+  return !userLocation ? (
```

Comment:
If MoonPay is down or doesn't return a location for a certain user (maybe because of their location?) they would get stuck on an infinite loading I believe. How would you feel about showing the loading only when the request is actually loading and if it fails for whatever reason just show all the providers? 

Maybe we could even add an extra message for clarity, but not sure if that's worth the effort.

Reply:
In the `useEffect`, I check if the API call had any type of error. If so, I use the phone number country code for location. Are you concerned that this "if clause" may not always be triggered?

Reply:
Um, not sure really, if the request times out, does the `status` still change (and hence the `useEffect` called)? If so that sounds enough, I just thought we could use the `loading` from the `useAsync` hook, but it might not be necessary 🤷‍♂️

Reply:
I think it's safe to assume a server timeout will be interpreted as an `error`. We could use `loading` as well but we'd probably need another state variable to make sure the `AcitivityIndicator` is showing before the `useAsync` hook is called.

Reply:
This ended up being resolved by your suggestion to contain all the logic within the `useAsync`!